app/src/main/java/vunt/com/vn/moviedb_28/data/source/remote/config/service/Api.java

```diff
@@ -1,4 +1,26 @@
 package vunt.com.vn.moviedb_28.data.source.remote.config.service;
 
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import retrofit2.http.GET;
+import retrofit2.http.Query;
+import vunt.com.vn.moviedb_28.data.source.remote.config.response.CategoryResult;
+import vunt.com.vn.moviedb_28.data.source.remote.config.response.GenreResult;
+
 public interface Api {
+
+    @GET("/3/movie/popular")
+    Single<CategoryResult> getPopular(@Query("api_key") String apiKey, @Query("page") int page);
```

Comment:
tất cả các api em đều cần viết lại câu lệnh @Querry("api_key")
em có thể tham khảo cách này để giảm thiểu số lần viết api key
https://futurestud.io/tutorials/retrofit-2-how-to-add-query-parameters-to-every-request

Reply:
chỗ này em sử dụng Interceptor ạ.